GAThread.java

```diff
@@ -0,0 +1,116 @@
+import java.util.concurrent.locks.ReentrantLock;
+import java.util.ArrayList;
+import java.util.HashMap;
+
+public class GAThread{
+   private int largeNumber = 3;
+   private int numFeatures = 6;
+   double[] weights = new double[numFeatures];
+   GeneticAlgorithm GA = new GeneticAlgorithm();
+   ArrayList<Genome> population = GA.initializePopulation();
+   private int blockadeWeightIndex = 0;
+   private int edgeHeightWeightIndex = 1;
+   private int heightDifferenceWeightIndex = 2;
+   private int holesWeightIndex = 3;
+   private int islandWeightIndex = 4;
+   private int parityWeightIndex = 5;
```

Comment:
We can also simply leave out the names of the indices since the only thing we need to know is the length of the weight vector. That should make the algorithm more general and simplify the crossover and mutation phase code into a single for-loop as well.

Reply:
Oh. I see. I can try this. Thanks.